src/main/kotlin/tech/thatgravyboat/skycubed/features/misc/CustomDamage.kt

```diff
@@ -0,0 +1,207 @@
+import net.minecraft.world.entity.decoration.ArmorStand
+import net.minecraft.world.phys.Vec3
+import tech.thatgravyboat.skyblockapi.api.events.base.Subscription
+import tech.thatgravyboat.skyblockapi.api.events.base.predicates.OnlyOnSkyBlock
+import tech.thatgravyboat.skyblockapi.api.events.entity.EntityAddedEvent
+import tech.thatgravyboat.skyblockapi.api.events.render.RenderWorldEvent
+import tech.thatgravyboat.skyblockapi.api.events.time.TickEvent
+import tech.thatgravyboat.skyblockapi.helpers.McFont
+import tech.thatgravyboat.skyblockapi.platform.drawString
+import tech.thatgravyboat.skyblockapi.utils.extensions.pushPop
+import tech.thatgravyboat.skyblockapi.utils.text.Text
+import tech.thatgravyboat.skyblockapi.utils.text.TextStyle.color
+import tech.thatgravyboat.skyblockapi.utils.time.currentInstant
+import tech.thatgravyboat.skycubed.config.rendering.RenderingConfig
+import tech.thatgravyboat.skycubed.features.misc.CustomDamage.DamageType.Companion.damageType
+import java.util.concurrent.ConcurrentLinkedQueue
+import kotlin.math.sin
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Instant
+
+@Module
+object CustomDamage {
+    val config = RenderingConfig.customDamage
+
+    private val damageRegex = Regex("([✧✯]?)([\\d,]+)+[✧✯]?([❤+⚔✷♞☄]?)")
+
+    private val damageList = ConcurrentLinkedQueue<Damage>()
+
+    @Subscription
+    @OnlyOnSkyBlock
+    fun onArmorStand(event: EntityAddedEvent) {
+        if (!config.enabled) return
+
+        val eventName = event.entity.customName ?: return
+
+        if (event.entity is ArmorStand) {
+            damageRegex.matchEntire(eventName.string)?.let {
+                val type = it.damageType(eventName.siblings[0]?.color)
+                val amount = (it.groups[2]?.value ?: return).replace(",", "").toLong()
+                val close = if (config.combining) damageList.firstOrNull { existing ->
+                    existing.type == type && event.entity.position().distanceTo(existing.position) < config.combineThreshold
+                } else null
+                if (close == null) {
+                    damageList.add(
+                        Damage(
+                            amount,
+                            type,
+                            currentInstant(),
+                            event.entity.position(),
+                            currentInstant()
+                        ),
+                    )
+                } else {
+                    close.amount += amount
+                    if (close.actualSpawnTime >= currentInstant() - config.fullTimeout.milliseconds) {
+                        close.time = currentInstant()
+                    }
+                }
+                event.entity.isCustomNameVisible = false
+            }
+        }
+    }
+
+    @Subscription(TickEvent::class)
+    @OnlyOnSkyBlock
+    fun onTick() {
+        damageList.removeIf { it.time < currentInstant() - config.timeout.milliseconds }
```

Comment:
```suggestion
        damageList.removeIf { it.time.since() > config.timeout.milliseconds }
```

i think this is the same, it needs an import though